receiver/prometheusreceiver/config.go

```diff
@@ -112,8 +110,11 @@ func (cfg *Config) Validate() error {
 }
 
 func (cfg *Config) validatePromConfig(promConfig *promconfig.Config) error {
-	if len(promConfig.ScrapeConfigs) == 0 && cfg.TargetAllocator == nil {
+	if (promConfig == nil || len(promConfig.ScrapeConfigs) == 0) && cfg.TargetAllocator == nil {
 		return errors.New("no Prometheus scrape_configs or target_allocator set")
```

Comment:
Consider to move this into the top `Validate` since this does not belong to the `promconfig.Config` but to the combination of that + the TargetAllocator.